From khali@linux-fr.org Sun Aug 13 14:40:13 2006
Date: Sun, 13 Aug 2006 23:40:09 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Petr Vandrovec <vandrove@vc.cvut.cz>
Subject: [PATCH 08/12] i2c-matroxfb: Struct init conversion
Message-Id: <20060813234009.1141081e.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-matroxfb-c99-struct-init.patch

i2c-matroxfb: Struct init conversion

Convert the struct i2c_algo_bit_data initialization to proper C99
style.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Acked-by: Petr Vandrovec <vandrove@vc.cvut.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/video/matrox/i2c-matroxfb.c |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

--- gregkh-2.6.orig/drivers/video/matrox/i2c-matroxfb.c
+++ gregkh-2.6/drivers/video/matrox/i2c-matroxfb.c
@@ -95,12 +95,13 @@ static struct i2c_adapter matrox_i2c_ada
 
 static struct i2c_algo_bit_data matrox_i2c_algo_template =
 {
-	NULL,
-	matroxfb_gpio_setsda,
-	matroxfb_gpio_setscl,
-	matroxfb_gpio_getsda,
-	matroxfb_gpio_getscl,
-	10, 10, 100,
+	.setsda		= matroxfb_gpio_setsda,
+	.setscl		= matroxfb_gpio_setscl,
+	.getsda		= matroxfb_gpio_getsda,
+	.getscl		= matroxfb_gpio_getscl,
+	.udelay		= 10,
+	.mdelay		= 10,
+	.timeout	= 100,
 };
 
 static int i2c_bus_reg(struct i2c_bit_adapter* b, struct matrox_fb_info* minfo, 
